src/shared/providers/fork-template-info/ForkTemplateInfoProvider.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, createContext } from 'react'
```

Comment:
THis should be a class - check for reference other Providers - functional components for Context sucks totally and creates performance issues - makes code harder to maintain because of a lot useMEmo and useCallback usage

Reply:
done